File: python/mpesa/utils/metrics.py
from typing import Any, Optional


class MetricsCollector:
    def increment(self, metric: str, tags: Optional[dict[str, str]] = None, value: int = 1) -> None:
        pass

    def gauge(self, metric: str, value: float, tags: Optional[dict[str, str]] = None) -> None:
        pass

    def timing(self, metric: str, duration_ms: float, tags: Optional[dict[str, str]] = None) -> None:
        pass

    def histogram(self, metric: str, value: float, tags: Optional[dict[str, str]] = None) -> None:
        pass


class NoopMetricsCollector(MetricsCollector):
    pass

File: go/types/metrics.go
package types

type MetricsCollector interface {
	Increment(metric string, tags map[string]string, value int)
	Gauge(metric string, value float64, tags map[string]string)
	Timing(metric string, durationMs float64, tags map[string]string)
	Histogram(metric string, value float64, tags map[string]string)
}

type NoopMetricsCollector struct{}

func (n *NoopMetricsCollector) Increment(_ string, _ map[string]string, _ int)     {}
func (n *NoopMetricsCollector) Gauge(_ string, _ float64, _ map[string]string)     {}
func (n *NoopMetricsCollector) Timing(_ string, _ float64, _ map[string]string)    {}
func (n *NoopMetricsCollector) Histogram(_ string, _ float64, _ map[string]string) {}
